cmd/chartsvc/handler.go

```diff
@@ -217,11 +217,17 @@ func newChartResponse(c *models.Chart) *apiResponse {
 }
 
 func newChartListResponse(charts []*models.Chart) apiListResponse {
+	chartDict := map[string]*apiResponse{}
 	cl := apiListResponse{}
 	for _, c := range charts {
-		cl = append(cl, newChartResponse(c))
+		chartResponse := newChartResponse(c)
+		// The same chart may be present in different repos, filter just one
+		id := (chartResponse.Relationships["latestChartVersion"].Data).(models.ChartVersion).Digest
```

Comment:
wouldn't it be simpler to just do

```suggestion
		id := c.ChartVersions[0].Digest
```
That way we avoid having to assert the type.

We can use `newChartResponse` below when we append to `cl`.